Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,387 @@
+        if tree.has_children(node):
+            left_child = tree.left_child(node)
+            right_child = tree.right_child(node)
+
+            w1, w2 = tree.weight(left_child), tree.weight(right_child)
+            alpha_1 = math.pi * w2 / (w1 + w2)
+            alpha_2 = math.pi * w1 / (w1 + w2)
+
+            length_1 = square.length * math.sin(alpha_1 / 2)
+            length_2 = square.length * math.sin(alpha_2 / 2)
+
+            angle_1 = square.angle + alpha_2 / 2
+            angle_2 = square.angle - alpha_1 / 2
+
+            center_1 = self._compute_center_left(
+                square, length_1, alpha_1, angle_1)
+            center_2 = self._compute_center_right(
+                square, length_2, alpha_2, angle_2)
+
+            square_1 = Square(center_1, length_1, angle_1)
+            square_2 = Square(center_2, length_2, angle_2)
+
+            children.append(self.pythagoras_tree(tree, left_child, square_1))
+            children.append(self.pythagoras_tree(tree, right_child, square_2))
+
+        return TreeNode(square, tree.parent(node), children)
+
+    def _compute_center_left(self, initial_square, length, alpha, angle):
+        """Compute the central point of the left child square.
+
+        Parameters
+        ----------
+        initial_square : Square
+            The sqare where we are continuing from.
+        length : float
+            The length of the side of the new square.
+        alpha : float
+            The angle at which the new square is offset from its parent.
+        angle : float
+            The computed angle at which the entire square is offset for
+            drawing.
+
+        Returns
+        -------
+        Point
+            The central point of the left square.
+
+        """
+        base_center, base_length, base_angle = initial_square
+        # get the point on the square side and prepare it for rotation
+        t0 = self._get_point_on_square_edge(
+            base_center, base_length, base_angle)
+        # get the edge point where we will rotate t0 around
+        square_diagonal_length = math.sqrt(2 * base_length ** 2)
+        edge = self._get_point_on_square_edge(
+            base_center, square_diagonal_length, base_angle + math.pi / 4)
+
+        # translate edge to origin, temp point is is copy of edge
+        temp = Point(edge.x - t0.x, edge.y - t0.y)
+        # rotate point around edge point and translate back
+        t1 = Point(
+            temp.x * math.cos(-alpha) - temp.y * math.sin(-alpha),
+            temp.x * math.sin(-alpha) + temp.y * math.cos(-alpha)
+        )
+        # translate rotated point back to its original spot
+        t1 = Point(t1.x + t0.x, t1.y + t0.y)
+
+        # calculate the middle point between the rotated point and edge
+        t2 = Point((t1.x + edge.x) / 2, (t1.y + edge.y) / 2)
+        return self._get_point_on_square_edge(t2, length, angle)
+
+    def _compute_center_right(self, initial_square, length, alpha, angle):
```

Comment:
The code for `_center_center_left()` and `_compute_center_right()` is almost the same. If you add another two parameters `op1` and `op2` (named appropriately) where they stand for `operator.add` and `operator.sub` (or vice versa), this can be a single function?